src/index.ts

```diff
@@ -9,30 +19,58 @@ const BROCCOLI_FEATURES = Object.freeze({
 
 const PATHS = new WeakMap();
 
-function isPossibleNode(node) {
-  let type = typeof node;
-
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function isPossibleNode(node: any): node is InputNode {
   if (node === null) {
     return false;
-  } else if (type === 'string') {
+  } else if (typeof node === 'string') {
     return true;
-  } else if (type === 'object' && typeof node.__broccoliGetInfo__ === 'function') {
+  } else if (typeof node === 'object' && typeof node.__broccoliGetInfo__ === 'function') {
     // Broccoli 1.x+
     return true;
-  } else if (type === 'object' && typeof node.read === 'function') {
+  } else if (typeof node === 'object' && typeof node.read === 'function') {
```

Comment:
I wonder if we should take the opportunity to kill this API?

Reply:
I think we may want to do a deprecation phase first, and then kill it?

Reply:
We've already been doing that in Broccoli builder for some time, and ember cli for some time too.
This will actually help with debugging when a bad plugin is used as the instantiation stack will be available unlike now where we don't have it and only know at build time.

Reply:
I think regardless of what we want to do we should do it in another PR since it isn't really related to the typescript conversion / this PR is already large.

Reply:
Agreed